quantum/impl/quantum_auxiliary_impl.h

```diff
@@ -0,0 +1,62 @@
+/*
+** Copyright 2022 Bloomberg Finance L.P.
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     http://www.apache.org/licenses/LICENSE-2.0
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+*/
+//NOTE: DO NOT INCLUDE DIRECTLY
+
+//##############################################################################################
+//#################################### IMPLEMENTATIONS #########################################
+//##############################################################################################
+
+namespace Bloomberg {
+namespace quantum {
+
+template <typename BitField>
+bool isIntersection(BitField lhs, BitField rhs)
+{
+    return static_cast<int>(lhs) & static_cast<int>(rhs);
```

Comment:
What if an enum is defined as 
enum class LongEnum : long long {...};
and isIntersection is called for two members of LongEnum?

Reply:
https://stackoverflow.com/a/28101901/19849344

I put a note, about conversion to int type
```
/// @note BitField type must be convertible to int.
```